autowiring/ContextCreatorBase.h

```diff
@@ -28,31 +28,17 @@ class ContextCreatorBase {
   /// </remarks>
   template<class Ctr, class Fx>
   void Clear(bool wait, Ctr& ctr, Fx&& locker) {
-    if(!wait) {
-      // Trivial signal-clear-return:
-      std::lock_guard<std::mutex> lk(m_contextLock);
-      for(const auto& ctxt : ctr) {
-        auto locked = locker(ctxt);
-        if(locked)
-          locked->SignalShutdown();
-      }
-      ctr.clear();
-      return;
+    // Copy out under lock
+    Ctr ctrCopy = (std::lock_guard<std::mutex>(m_contextLock), std::move(ctr));
```

Comment:
Recommend you change `ctrCopy` to `ctrLocal`, and update your comment to reflect the fact that what's happening here is a move, not a copy.